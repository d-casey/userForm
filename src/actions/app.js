import {
  RESET_FORM_STAGE,
  UPDATE_STAGE
} from '../constants/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})
