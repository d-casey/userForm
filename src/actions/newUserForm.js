import {
  RESET_USER_FORM,
  SUBMIT_NEW_USER_FORM
} from '../constants/actions'

export const submitNewUserForm = formFields => ({
  type: SUBMIT_NEW_USER_FORM,
  payload: formFields
})

export const resetUserForm = () => ({
  type: RESET_USER_FORM
})
