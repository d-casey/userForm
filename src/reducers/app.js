import {
  UPDATE_STAGE,
  RESET_FORM_STAGE
} from '../constants/actions'

const initialState = {
  stage: 0
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

      case RESET_FORM_STAGE:
        return {
          ...state,
          stage: 0
        }

    default:
      return state
  }
}
