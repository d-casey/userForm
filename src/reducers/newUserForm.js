import {
  SUBMIT_NEW_USER_FORM,
  RESET_USER_FORM
} from '../constants/actions'

const initialState = {
  name: "",
  role: "",
  email: "",
  password: "",
  isSubmitted: false
}

export default function newUserForm(state = initialState, action) {
  switch (action.type) {
    case SUBMIT_NEW_USER_FORM:
      return {
        ...state,
        name: action.payload.name,
        role: action.payload.role,
        email: action.payload.email,
        password: action.payload.password,
        isSubmitted: true
      }

    case RESET_USER_FORM:
      return {
        ...state,
        name: "",
        role: "",
        email: "",
        password: "",
        isSubmitted: false
      }

    default:
      return state
  }
}
