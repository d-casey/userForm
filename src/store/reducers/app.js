import {
  UPDATE_STAGE,
  RESET_FORM_STAGE,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

const initialState = {
  stage: 0,
  formCompleted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

    case RESET_FORM_STAGE:
      return {
        ...state,
        stage: 0
      }

    case SET_FORM_COMPLETION:
      return {
        ...state,
        formCompleted: true
      }

    case RESET_APP:
      return {
        ...state,
        formCompleted: false,
        stage: 0
      }

    default:
      return state
  }
}
