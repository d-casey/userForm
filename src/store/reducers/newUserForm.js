import {
  UPDATE_USER_FIELDS,
  RESET_USER_FORM
} from '../constants/actions'

export const initialState = {
  name: "",
  role: "",
  email: "",
  password: "",
  isSubmitted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_USER_FIELDS:
      return {
        ...state,
        name: action.payload.name,
        role: action.payload.role,
        email: action.payload.email,
        password: action.payload.password,
        isSubmitted: true
      }

    case RESET_USER_FORM:
      return initialState

    default:
      return state
  }
}
