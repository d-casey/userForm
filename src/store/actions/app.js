import {
  RESET_FORM_STAGE,
  UPDATE_STAGE,
  LOGGING_DETAILS,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

import { resetUserForm, resetPrivacyForm } from '../../store/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})

export const resetAllForms = () => dispatch => {
  dispatch(resetUserForm())
  dispatch(resetPrivacyForm())
}

export const setFormCompletion = () => ({
  type: SET_FORM_COMPLETION
})

export const resetApp = () => ({
  type: RESET_APP
})

export const loggingDetails = () => ({
  type: LOGGING_DETAILS
})

export const logSubmittedFormDetails = (formDetails) => dispatch => {
  dispatch(loggingDetails())
  console.log('formDetails: ', formDetails)
  dispatch(resetAllForms())
  dispatch(setFormCompletion())
}
