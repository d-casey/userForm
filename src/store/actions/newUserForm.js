import {
  RESET_USER_FORM,
  UPDATE_USER_FIELDS
} from '../constants/actions'

import { updateStage } from '../actions'

export const updateUserFields = formFields => ({
  type: UPDATE_USER_FIELDS,
  payload: formFields
})

export const submitNewUserForm = formFields => dispatch => {
  dispatch(updateUserFields(formFields))
  dispatch(updateStage(1))
}

export const resetUserForm = () => ({
  type: RESET_USER_FORM
})
