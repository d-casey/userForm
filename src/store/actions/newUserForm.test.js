import * as appActions from './app'
import * as newUserFormActions from './newUserForm'

describe('newUserForm actions', () => {
  describe('updateUserFields', () => {
    it('sends the form fields as the payload', () => {
      expect(newUserFormActions.updateUserFields({ formFields: {} })).toMatchSnapshot()
    })
  })

  describe('resetUserForm', () => {
    it('sends the resetUserForm actions', () => {
      expect(newUserFormActions.resetUserForm()).toMatchSnapshot()
    })
  })

  describe('submitNewUserForm', () => {
    let dispatch, formFields
    beforeEach(() => {
      dispatch = jest.fn()
      formFields = 'FORM'
      newUserFormActions.submitNewUserForm(formFields)(dispatch)
    })

    test('it calls dispatch two times', () => {
      expect(dispatch).toHaveBeenCalledTimes(2)
    })

    it('calls dispatch with the updateUserFields action', () => {
      expect(dispatch).toBeCalledWith({'payload': 'FORM', 'type': 'UPDATE_USER_FIELDS'})
    })

    it('calls dispatch to update the stage', () => {
      expect(dispatch).toBeCalledWith({'payload': {'stage': 1}, 'type': 'UPDATE_STAGE'})
    })
  })
})
