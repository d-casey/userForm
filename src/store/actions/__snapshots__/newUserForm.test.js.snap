// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newUserForm actions resetUserForm sends the resetUserForm actions 1`] = `
Object {
  "type": "RESET_USER_FORM",
}
`;

exports[`newUserForm actions updateUserFields sends the form fields as the payload 1`] = `
Object {
  "payload": Object {
    "formFields": Object {},
  },
  "type": "UPDATE_USER_FIELDS",
}
`;
